node_modules/gh-users/tests/test-terms-and-conditions.js

```diff
@@ -0,0 +1,296 @@
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+var _ = require('lodash');
+var assert = require('assert');
+
+var ConfigTestsUtil = require('gh-config/tests/util');
+var SeriesTestsUtil = require('gh-series/tests/util');
+var TestsUtil = require('gh-tests/lib/util');
+
+var UsersTestsUtil = require('gh-users/tests/util');
+
+describe('Terms and Conditions', function() {
+
+    /**
+     * Create a tenant and application that contains 2 users and has an enabled Terms and Conditions
+     *
+     * @param  {Function}       callback                            Standard callback function
+     * @param  {Tenant}         callback.tenant                     The created tenant
+     * @param  {App}            callback.app                        The created application
+     * @param  {RestClient}     callback.globalAdminClient          A global admin rest client
+     * @param  {Object}         callback.user1                      A regular user in the application
+     * @param  {Object}         callback.user2                      A regular user in the application
+     * @api private
+     */
+    var setup = function(callback) {
+        TestsUtil.generateTestTenant(1, function(tenant, app) {
+            TestsUtil.generateTestUsers(app, 2, false, function(simon, nico) {
+
+                // Sanity-check that the application has no Terms and Conditions
+                UsersTestsUtil.assertGetMe(simon.client, function(me) {
+                    assert.strictEqual(me.termsAndConditions.needsToAccept, false);
+
+                    // Set some Terms and Conditions
+                    TestsUtil.getGlobalAdminRestClient(function(globalAdminClient) {
+                        var configuration = {
+                            'allowLocalAccountCreation': true,
+                            'termsAndConditionsEnabled': true,
+                            'termsAndConditionsText': 'Oh lawdy'
+                        };
+                        ConfigTestsUtil.assertUpdateConfig(globalAdminClient, app.id, configuration, function() {
+                            return callback(tenant, app, globalAdminClient, simon, nico);
+                        });
+                    });
+                });
+            });
+        });
+    };
+
+    describe('System interaction', function() {
+
+        /**
+         * Test that verifies that user need to accept the Terms and Conditions when creating an account
+         */
+        it('verify users need to accept the Terms and Conditions when creating an account', function(callback) {
```

Comment:
Can admins create accounts on behalf of users? If so, they should be able to create a user account while deferring acceptance of the T&C to when the user attempts to use the system I would think. That should also be tested.